src/transformers/models/swin2sr/image_processing_swin2sr.py

```diff
@@ -0,0 +1,182 @@
+    model_input_names = ["pixel_values"]
+
+    def __init__(
+        self,
+        do_rescale: bool = True,
+        rescale_factor: Union[int, float] = 1 / 255,
+        do_pad: bool = True,
+        pad_size: int = 8,
+        **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.do_rescale = do_rescale
+        self.rescale_factor = rescale_factor
+        self.do_pad = do_pad
+        self.pad_size = pad_size
+
+    def rescale(
+        self, image: np.ndarray, scale: float, data_format: Optional[Union[str, ChannelDimension]] = None, **kwargs
+    ) -> np.ndarray:
+        """
+        Rescale an image by a scale factor. image = image * scale.
+
+        Args:
+            image (`np.ndarray`):
+                Image to rescale.
+            scale (`float`):
+                The scaling factor to rescale pixel values by.
+            data_format (`str` or `ChannelDimension`, *optional*):
+                The channel dimension format for the output image. If unset, the channel dimension format of the input
+                image is used. Can be one of:
+                - `"channels_first"` or `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
+                - `"channels_last"` or `ChannelDimension.LAST`: image in (height, width, num_channels) format.
+
+        Returns:
+            `np.ndarray`: The rescaled image.
+        """
+        return rescale(image, scale=scale, data_format=data_format, **kwargs)
+
+    def pad(self, image: np.ndarray, size: int, data_format: Optional[Union[str, ChannelDimension]] = None):
+        """
+        Pad an image to make the height and width divisible by `size`.
+
+        Args:
+            image (`np.ndarray`):
+                Image to pad.
+            size (`int`):
+                The size to make the height and width divisible by.
+            data_format (`str` or `ChannelDimension`, *optional*):
+                The channel dimension format for the output image. If unset, the channel dimension format of the input
+                image is used. Can be one of:
+                - `"channels_first"` or `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
+                - `"channels_last"` or `ChannelDimension.LAST`: image in (height, width, num_channels) format.
+
+        Returns:
+            `np.ndarray`: The rescaled image.
+        """
+        image = to_channel_dimension_format(image, ChannelDimension.LAST)
+        old_height, old_width = image.shape[0], image.shape[1]
+
+        pad_height = (old_height // size + 1) * size - old_height
+        pad_width = (old_width // size + 1) * size - old_width
+        image = np.concatenate([image, np.flip(image, [0])], 0)[: old_height + pad_height, :, :]
+        image = np.concatenate([image, np.flip(image, [1])], 1)[:, : old_width + pad_width, :]
+
+        if data_format is not None:
+            image = to_channel_dimension_format(image, data_format)
+
+        return image
```

Comment:
cc @amyeroberts I've implemented padding as follows to match the original implementation, let me know if that's ok.

i'm not using pad from image_transforms atm